index.md

```diff
@@ -17,24 +17,160 @@ redirect_from:
+not cause fragmentation and by allowing DNS to switch from UDP to TCP when
+larger buffer sizes are not enough.
+
+Note: EDNS buffer size
+----------------------
+
+**Please note** that the _exact recommended EDNS buffer sizes_ have not been
+agreed upon, the current ballpark around 1200 (1220, 1232, ...) is to limit the risk
+of fragmentation in IPv6.
+If you have comments or suggestion then please join the discussion at
+[dns-operations](https://lists.dns-oarc.net/mailman/listinfo/dns-operations)
+mailing list.
+
+Action: Authoritative DNS Operators
+-----------------------------------
+
+For the authoritative side what you should do to help with these issues
+is to **_honor [RFC 7766](https://tools.ietf.org/html/rfc7766)_** and
+answer DNS queries over TCP (port 53), _check your firewall(s) also!_
+
+You should also use an EDNS buffer size that will not cause fragmentation,
+recommended here is around 1220 bytes but it is still up for discussion.
+
+And lastly, _Authoritative **MUST NOT** send oversized answers!_
+
+Action: DNS Resolver Operators
+------------------------------
+
+For the resolver side it's more or less the same requirement as for the
+authoritative, **_honor [RFC 7766](https://tools.ietf.org/html/rfc7766)_**
+_(DNS over TCP port 53)_ and use an EDNS buffer size _(~1220 bytes)_ that
+will not cause fragmentation. _Remember to check your firewall(s)!_
+
+And for that last important part, _Resolvers **MUST** repeat queries over
+TCP if they receive a truncated UDP response (with TC=1 set)!_
+
+Action: DNS software vendors
+----------------------------
+
+As a DNS software vendor it is important to be **standards compliant** and
+to use a _**default EDNS buffer size** (~1220)_ that will not cause
 fragmentation.
 
-Please subscribe to [mailing list dns-announce](https://lists.dns-oarc.net/mailman/listinfo/dns-announce)
-or follow [dnsflagday Twitter](https://www.twitter.com/dnsflagday)
-to receive notification when more information becomes available.
+How to test?
+------------
+
+We are working on a web based testing tool and once it's ready you will
+find it on this page, until then you can test this by using the
+following CLI commands:
+
+```shell
+$ dig +tcp @auth_IP yourdomain.example.
+$ dig +tcp @resolver_IP yourdomain.example.
+$ dig @resolver_IP test.knot-resolver.cz. TXT
+```
+
+If you are a service provider you can also test your authoritative and
+resolver services by allowing DNS over TCP and changing the configuration
+for the default EDNS buffer size:
+
+- BIND
+```
+options { edns-udp-size 1220; };
+```
+
+- Knot DNS
+```
+net.bufsize(1220)
+```
+
+- Knot Resolver
+```
+net.bufsize(1220)
+```
+
+- PowerDNS Authoritative
+```
+udp-truncation-threshold=1220
+```
+
+- PowerDNS Recursor
+```
+edns-outgoing-bufsize=1220
+udp-truncation-threshold=1220
+```
+
+- Unbound
+```
+server:
+    edns-buffer-size: 1220
+```
+
+- NSD
+```
+server:
+    ipv4-edns-size: 1200
+    ipv6-edns-size: 1200
+```
```

Comment:
Is 1200 intentional when 1220 is everywhere else?

Reply:
That was what the NSD people added, ping @bjovereinder

Reply:
not intentional, should be 1220 indeed